internal/unionstore/memdb.go

```diff
@@ -328,6 +328,12 @@ func (db *MemDB) set(key []byte, value []byte, ops ...kv.FlagsOp) error {
 		return nil
 	}
 
+	// If this node is newly allocated and the value length is greater than zero, add
+	// newly inserted flag to it.
+	if len(value) > 0 && isNewNode {
+		x.setKeyFlags(kv.ApplyFlagsOps(x.getKeyFlags(), kv.SetNewlyInserted))
```

Comment:
I notice that the flags in memdb is not rollbackable. Then, is it possible that persistent flags may be reset incorrectly?

For example, `SetNewlyInserted` is set already. Then, we set the key again, but just in a staging state. Later, we rollback the staging state, will `SetNewlyInserted` be cleared unexpectedly?